sgkit/stats/grm.py

```diff
@@ -32,18 +32,60 @@ def _grm_VanRaden(
     return G
 
 
+def _grm_Endelman_Jannink(
+    call_dosage: ArrayLike,
+    ancestral_frequency: ArrayLike,
+    ploidy: int,
+    skipna: bool = False,
+) -> ArrayLike:
+    if skipna:
+        raise NotImplementedError(
+            "The 'skipna' option is not implemented for the 'Endelman-Jannink' estimator"
+        )
+    m, n = call_dosage.shape
+    ancestral_dosage = ancestral_frequency * ploidy
+    W = call_dosage - ancestral_dosage[:, None]
+    W_mean = da.nanmean(W, axis=0, keepdims=True)
+    Z = W - W_mean
+    S = (Z.T @ Z) / m
+    T = da.diag(S).mean() * da.eye(n)
+    # calculate scaling parameter delta
+    Z2 = Z**2
+    Gama = Z2.T @ Z2 / m  # E&J eq 25
+    delta_numerator = (Gama - S**2).sum() / m  # Error in E&J eq 24?
+    delta_denominator = ((S - T) ** 2).sum()  # squared Frobenius norm
+    delta = delta_numerator / delta_denominator  # E&J eq 20
+    # delta must be in [0 1]
+    delta = da.maximum(delta, 0.0)
+    delta = da.minimum(delta, 1.0)
+    # E&J eq 17
+    numerator = (T * delta + (1 - delta) * S) + W_mean.T @ W_mean
+    denominator = ploidy * da.mean(ancestral_frequency * (1 - ancestral_frequency))
+    G = numerator / denominator
+    return G
+
+
 def genomic_relationship(
     ds: Dataset,
     *,
     call_dosage: Hashable = variables.call_dosage,
-    estimator: Optional[Literal["VanRaden"]] = None,
+    estimator: Optional[Literal["VanRaden", "Endelman-Jannink"]] = None,
```

Comment:
I'm a bit nervous about how many times we're using the literal strings here, and the potential for easy-to miss errors ("like 'Endelman-jannink' or something).

Would it make sense to have something like
```
EST_VAN_RADEN = "VanRaden"
EST_ENDELMAN_JANNICK = "Endelman-Jannink"

def genomic_relationship(...
    estimator: Optional[Literal[EST_VAN_RADEN, EST_ENDELMAN_JANNINK]]   
```
and use the constants in the following code? That seems like a reasonable compromise between readability and reducing chances for error, without getting into a tedious global constant declaration thing?